src/pools/abstract-pool.ts

```diff
@@ -202,14 +214,24 @@ export abstract class AbstractPool<
       throw new Error(
         'Cannot instantiate a pool with a negative number of workers'
       )
-    } else if (!this.dynamic && numberOfWorkers === 0) {
+    } else if (this.type === PoolType.FIXED && numberOfWorkers === 0) {
       throw new Error('Cannot instantiate a fixed pool with no worker')
     }
   }
 
+  private checkPoolOptions (opts: PoolOptions<Worker>): void {
+    this.opts.enableEvents =
+      opts.enableEvents !== undefined || opts?.enableEvents !== null
```

Comment:
This can be simplified to `opts.enableEvents != null`

Reply:
Or I think it also can be shortened to `this.opts.enableEvents = opts.enableEvents ?? true`

Reply:
The goal here is the check and affect the value given at objet instantiation.

Reply:
Yeah I know, it the same but just smaller syntax
The `??` operator is really new to JS, you should read about optional coalescing
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator

Reply:
@jerome-benoit Did you see my last response here?

Reply:
I guess not :)